### DISCUSSION TITLE
Automatic-Module-Name: org.jgroups gone in 5.4.0.Final

### DISCUSSION THREAD

**Discussion by eitan-rosenberg** [belaban/JGroups#869] [General]
The MANIFEST.MF in jgroups-5.3.13.Final.jar use to have:
**Automatic-Module-Name: org.jgroups** 
Now in jgroups-5.4.0.Final.jar it is gone.
Can it be put back ?
I think it can be done using <artifactId>maven-jar-plugin</artifactId>

**Comment by belaban** [comment]
How? Please create a JIRA with instructions (or a PR)...

**Reply by eitan-rosenberg** [reply]
I am not this kind of expert...

Usually use Eclipse to do this. 

I read about it here: 
https://maven.apache.org/plugins/maven-jar-plugin/examples/manifest-customization.html

So: 
You already use "maven-jar-plugin" in your pom.xml
![image](https://github.com/user-attachments/assets/92f5e657-263f-433e-8a34-18e0f8fe163f)

**I did a small test and add manifest Entries**

			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-jar-plugin</artifactId>
				<configuration>
					<archive>
						<manifest>
							<addDefaultEntries>true</addDefaultEntries>
							<addDefaultImplementationEntries>true</addDefaultImplementationEntries>
							<addBuildEnvironmentEntries>true</addBuildEnvironmentEntries>
							
							
							<mainClass>org.jgroups.Version</mainClass>
						</manifest>
						<manifestEntries>
							<Automatic-Module-Name>org.jgroups</Automatic-Module-Name>
						</manifestEntries>
					</archive>
				</configuration>
			</plugin>
 
**The result of "mvn package"** 
![image](https://github.com/user-attachments/assets/214f7423-3562-4cb5-8068-6e1278097113)

I hope that you find this helpful.

**Comment by eitan-rosenberg** [comment]
Found this
https://github.com/belaban/JGroups/discussions/688

**Comment by belaban** [comment]
I don't see `Automatic-module-name` in `pom.xml` anymore, do I need to re-add it?

**Reply by eitan-rosenberg** [reply]
Sure 
You did that 2 years ago, see here https://github.com/belaban/JGroups/discussions/688
I wonder what happens between 5.3.13 and 5.4.0
**This is not urgent**

**Comment by eitan-rosenberg** [comment]
Thanks for 5.4.1.Final
<Automatic-Module-Name>org.jgroups</Automatic-Module-Name> is back in town :-)
Have a nice weekend and merry Christmas.

**Comment by belaban** [comment]
Thanks, same to you!
